src/client.rs

```diff
@@ -1104,11 +1130,16 @@ impl<T: Read + Write> Session<T> {
             .map(|f| f.to_string())
             .collect::<Vec<String>>()
             .join(" ");
+        let datestr = match date {
+            Some(date) => format!(" \"{} +0000\"", date.format("%d-%h-%Y %T")),
```

Comment:
It seems odd to enforce a timezone of `+0000` here? I think I'd prefer for this method to take a normal `DateTime` (i.e., with timezone information), and then for that timezone to also be used in the datestr (in which case I think you can just use `date.format` and don't need the `format!` as well.

Reply:
Yeah, it makes sense to switch to DateTime.

Regarding the use of `format!` is to swallow the extra space and double quotes in case of `None` for date. I know that they can be included in the date.format I just opted for clarity on what the date format is vs how it fits on the APPEND line.

Reply:
I've updated it now with `DateTime<Utc>`. I'm really new to Rust so still getting the hang of the basic tool belt and language features. I'm guessing that `Utc` should not be hard-coded but specified with generics? E.g. 

```rust
    pub fn append_with_flags_and_date<S: AsRef<str>, B: AsRef<[u8]>, Z: TimeZone>(
        &mut self,
        mailbox: S,
        content: B,
        flags: &[Flag<'_>],
        date: Option<DateTime<Z>>,
    ) -> Result<()> {
```

However, this trickles up the chain to the other append methods in a way I don't quite understand yet:

![image](https://user-images.githubusercontent.com/247048/102003116-2ef4e480-3d3e-11eb-9869-212093df2309.png)

Reply:
I tried a bunch of variants as `Option<DateTime<TimeZone<offset = O>>>` (and one with `dyn` as well) but cannot seem to get it right.

Reply:
Changed DateTime<Local> to DateTime<FixedOffset>

Local takes the timezone from the system. Utc sets to +0000 we seem to
be wanting one that remains fixed so that when formatted it is a
specified.

Reply:
`date: Option<impl Into(DateTime<FixedOffset>)>` sounds great to me! Though I think you can move the `Into` one level out:
```rust
date: impl Into<Option<DateTime<FixedOffset>>>
```